crates/ra_ide_api/src/hover.rs

```diff
@@ -77,34 +78,177 @@ impl HoverResult {
+                    res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                }
+                hir::ImplItem::Const(it) => {
+                    let it = it.source(db).1;
+                    res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                }
+                hir::ImplItem::TypeAlias(it) => {
+                    let it = it.source(db).1;
+                    res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                }
+            },
+            Some(Def(it)) => {
+                match it {
+                    hir::ModuleDef::Module(it) => {
+                        let it = it.definition_source(db).1;
+                        if let hir::ModuleSource::Module(it) = it {
+                            res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                        }
+                    }
+                    hir::ModuleDef::Function(it) => {
+                        let it = it.source(db).1;
+                        res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                    }
+                    hir::ModuleDef::Struct(it) => {
+                        let it = it.source(db).1;
+                        res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                    }
+                    hir::ModuleDef::Union(it) => {
+                        let it = it.source(db).1;
+                        res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                    }
+                    hir::ModuleDef::Enum(it) => {
+                        let it = it.source(db).1;
+                        res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                    }
+                    hir::ModuleDef::EnumVariant(it) => {
+                        let it = it.source(db).1;
+                        res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                    }
+                    hir::ModuleDef::Const(it) => {
+                        let it = it.source(db).1;
+                        res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                    }
+                    hir::ModuleDef::Static(it) => {
+                        let it = it.source(db).1;
+                        res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                    }
+                    hir::ModuleDef::Trait(it) => {
+                        let it = it.source(db).1;
+                        res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                    }
+                    hir::ModuleDef::TypeAlias(it) => {
+                        let it = it.source(db).1;
+                        res.extend(hover_text(it.doc_comment_text(), it.short_label()))
+                    }
+                    hir::ModuleDef::BuiltinType(_) => {
+                        // FIXME: hover for builtin Type ?
+                    }
```

Comment:
We really should start to expose more hir methods as traits to avoid such boilerplate impls...